src/tickit_devices/eiger/eiger_adapters.py

```diff
@@ -430,5 +430,5 @@ def __init__(self, device: EigerDevice) -> None:
 
     def after_update(self) -> None:
         """Updates IOC values immediately following a device update."""
-        buffered_data = self.device.stream.consume_data()
-        self.add_message_to_stream([list(buffered_data)])
+        if buffered_data := list(self.device.stream.consume_data()):
```

Comment:
This is a good patch but I think the proper fix is in the ZMQ adapter itself, have made an issue: https://github.com/dls-controls/tickit/issues/196